scripts/bump_major_version.py

```diff
@@ -0,0 +1,335 @@
+                           version_string_name(from_version))
+    dest_dir = os.path.join(docs_directory, '_includes',
+                            version_string_name(to_version))
+
+    if os.path.exists(dest_dir):
+        shutil.rmtree(dest_dir)
+    shutil.copytree(src_dir, dest_dir)
+
+    # Traverse down the directory containing the "include files", adding each
+    # to our list of files to check later for references to the old version.
+    marktree(dest_dir)
+
+
+def copy_images(from_version, to_version):
+    # Int Int -> IO!
+    src_dir = os.path.join(docs_directory, 'images',
+                           version_string_name(from_version))
+    dest_dir = os.path.join(docs_directory, 'images',
+                            version_string_name(to_version))
+    if os.path.exists(dest_dir):
+        shutil.rmtree(dest_dir)
+    shutil.copytree(src_dir, dest_dir)
+
+
+def copy_sidebar_json(from_version, to_version):
+    # Int Int -> IO! State!
+    src_version = version_string_name(from_version)
+    dest_version = version_string_name(to_version)
+
+    # Do it once for CockroachDB
+    src_json = os.path.join(docs_directory,
+                            '_includes',
+                            'sidebar-data-' + src_version + '.json')
+    dest_json = os.path.join(docs_directory,
+                             '_includes',
+                             'sidebar-data-' + dest_version + '.json')
+
+    shutil.copy2(src_json, dest_json)
+    files_to_check[dest_json] = 1
+
+    # Do it once more for CockroachCloud
+    src_json_cloud = os.path.join(docs_directory,
+                                  '_includes',
+                                  'sidebar-data-' + src_version +
+                                  '.cockroachcloud.json')
+    dest_json_cloud = os.path.join(docs_directory,
+                                   '_includes',
+                                   'sidebar-data-' + dest_version +
+                                   '.cockroachcloud.json')
+
+    shutil.copy2(src_json_cloud, dest_json_cloud)
+    files_to_check[dest_json_cloud] = 1
+
+
+def strip_out_new_or_changed_in_vXYZ(version, file):
+    # Int Path -> IO!
+
+    # ugggggggghhhhhh this should become a regex, but for now this is serving
+    # as "documentation" of all the variant usages.
+    new_in_vXXX_string_00 = '<span class="version-tag">New in ' + \
+        version_string_name(version) + ':</span>'
+    new_in_vXXX_string_01 = '<span class="version-tag">New in ' + \
+        version_string_name(version) + '</span>:'
+    new_in_vXXX_string_02 = '<span class="version-tag">New in ' + \
+        version_string_name(version) + '</span>'
+    changed_in_vXXX_string_00 = '<span class="version-tag">Changed in ' + \
+        version_string_name(version) + '</span>'
+    changed_in_vXXX_string_01 = '<span class="version-tag">Changed in ' + \
+        version_string_name(version) + ':</span>'
+
+    # Read in the file
```

Comment:
Since we repeat this chunk and the "Write the file out again" chunk quite a bit, they might make sense as standalone functions to be called.